roofit/roofitcore/src/RooDataHist.cxx

```diff
@@ -1142,6 +1142,50 @@ RooPlot *RooDataHist::plotOn(RooPlot *frame, PlotOpt o) const
   return RooAbsData::plotOn(frame,o) ;
 }
 
+
+void RooDataHist::weights(double* output, size_t nEvents, RooAbsArg* obsList, RooFit::Detail::DataMap const& dataMap, bool correctForBinSize) 
+{
+  auto xVals = dataMap.at(obsList);
+
+  RooAbsBinning& binning = static_cast<RooRealVar*>(obsList)->getBinning();
+  bool isUniformBinning = dynamic_cast<RooUniformBinning const*>(&binning);
+
+  if (isUniformBinning) {
```

Comment:
```suggestion
  if (binning.isUniform()) {
```
`dynamic_cast` is expensive. Fortunately, the `RooAbsBinning` has a member function that gives you the information you need!